projects/common/lib/services/page-state.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from '@angular/core';
+import { Route } from '@angular/router';
+
+export interface PageList {
+  index: number;
+  path: string;
+  isComplete: boolean;
+  [key: string]: any;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PageStateService {
+
+  pageList: PageList[] = [];
+
+  constructor() { }
+
+  setPages( arr: Route[],
+            routeListConst: any,
+            fn?: (x: any) => {[key: string]: any} ): PageList[] {
+
+    if ( this.pageList && !this.pageList.length ) {
+
+      const routeConst = Object.keys( routeListConst ).map( x => routeListConst[x] );
+
+      let cnt = 0;
+      this.pageList = arr.filter((itm: any) => !itm.redirectTo)
+        .map((itm: any) => {
```

Comment:
You can delete your `cnt` variable.

The "map" method on an array has multiple parameters, one of them includes an index.  So you can just use that.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

In practice, I think you'd have to change line 30 to (code is untested):

```suggestion
        .map((itm: any, index: number) => {
```